src/embed/VbrickEmbed.ts

```diff
@@ -35,21 +35,37 @@ export abstract class VbrickEmbed implements IVbrickBaseEmbed {
 		this.eventBus = new EventBus(this.iframe, this.config);
 		this.initializeEmbed();
 
-		this.init = Promise.all([
+		return this.init = Promise.all([
 			this.initializeToken(),
-			this.eventBus.awaitEvent('load', 'error')
-		])
-			.then(([token])=> {
-				this.logger.log('embed loaded, authenticating');
-				this.eventBus.publish('authenticated', { token });
-			})
-			.catch(err => {
-				this.logger.error('Embed initialization error: ', err);
-
-				this.eventBus.publishError('Error loading embed content', err);
+			this.eventBus.awaitEvent('load')
+		]).then(([token])=> {
+			if (!token) {
+				return this.logger.log('embed loaded');
+			}
+			this.logger.log('embed loaded, authenticating');
+			this.eventBus.publish('authenticated', { token });
+
+			// added fail-safe check for if authChanged event isn't passed
+			// COMBAK - remove when confirmed unnecessary
+			const cleanupCallbacks: (() => void)[] = [];
+			const whenLoaded = new Promise<void>((resolve) => {
+				cleanupCallbacks.push(
+					this.eventBus.on('videoLoaded', resolve),
+					this.eventBus.on('webcastLoaded', resolve)
+				);
```

Comment:
Don't listen to videoLoaded or webcastLoaded here. What case are you trying to solve?